enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -574,5 +574,200 @@ describeWithToken("formatting > sandboxes", () => {
         "not.exist",
       );
     });
+
+    it.skip("advanced sandboxing based on saved question with joins should allow sandboxed user to use joins (metabase-enterprise#524)", () => {
+      const COLLECTION_GROUP_ID = 4;
+      const [ORDERS_ALIAS, PRODUCTS_ALIAS, REVIEWS_ALIAS] = [
+        "Orders",
+        "Products",
+        "Reviews",
+      ];
+
+      withSampleDataset(
+        ({
+          ORDERS,
+          ORDERS_ID,
+          PEOPLE,
+          PEOPLE_ID,
+          PRODUCTS,
+          PRODUCTS_ID,
+          REVIEWS,
+          REVIEWS_ID,
+        }) => {
+          cy.log(
+            "**-- 1. Create question based on repro steps in [#524](https://github.com/metabase/metabase-enterprise/issues/524) --**",
+          );
+
+          cy.request("POST", "/api/card", {
+            name: "EE_524",
+            dataset_query: {
+              database: 1,
+              query: {
+                joins: [
```

Comment:
All these joins in the sandbox question was actually `strategy: "inner-join"`, but it's not going to change the error, so you can ignore that. I just added those, since it was more reflective of the initial report ticket (and from a logical point).

Reply:
Good catch. I forgot to change them in UI when I was "spying" on console to see which query will it produce.

Reply:
Changed to inner-join since I was already changing some other things in this repro.